app/main.py

```diff
@@ -1,3 +1,24 @@
+import functools
+
+
 def cache(func):
-    # Write your code here
-    pass
+    global_cache = dict({})
```

Comment:
The cache creates a unique for each function because it is placed inside the decorator.